### PR TITLE
Add basic support for the PA31 delta file format (Fixes #1)

### LINKED ISSUE #1: Support for newer PA31 format
The [Windows 11 Insider Preview 10.0.26052.1100 (ge_release) amd64](https://uupdump.net/selectlang.php?id=73776830-7a01-481e-96b5-d1035cd617c5) update package comes with delta files which aren't supported by DeltaDownloader. The new delta files have the `PA31` signature, while DeltaDownloader only supports `PA30`:
https://github.com/Wack0/DeltaDownloader/blob/ab71359fc5a1f2446b650b31450c74a701c40979/MsDelta/DeltaFile.cs#L24

Here's an example of such delta file:
[pclxl.zip](https://github.com/Wack0/DeltaDownloader/files/14230341/pclxl.zip)
image

### PR DESCRIPTION
**Pull request by m417z**

### CODE REVIEW

**Comment by Wack0** [comment]
Thanks for the PR.

I just reversed new UpdateCompression.dll (seems they reimplemented everything natively for speed; msdelta of course uses some weird scripting language)

From static analysis, here is what I've determined the new header elements to be:

````
Int HeaderInfoSize (don't really care about this? for PA31 v1 it's 0x88)
Int IsPa31 (must be 1 for PA31, if 0 then UpdateCompression!DeltaPatch::CreatePatch creates a PA30)
Int DeltaClientMinVersion (must be == 1 currently for PA31)
Buffer AdditionalHash (not sure from just static analysis. hash of source file maybe? the same hash algorithm as the other hash is used)
````

These elements (or the equivalent in new `DELTA_HEADER_INFO_EX` structure, sizeof=0x88) are set to some default values when reading a PA30:
````
HeaderInfoSize = sizeof(DELTA_HEADER_INFO); // 0x50
IsPa31 = 0
DeltaClientMinVersion = 0
AdditionalHash.length = 0
````

Could you update the PR to include this? (if you could actually test that my guess of AdditionalHash is indeed correct, that would be nice too)

File: MsDelta/DeltaFile.cs [hunk 1]
```diff
@@ -10,38 +10,84 @@ namespace MsDelta
...
-            while (!HasSignatureAtOffset(bytes, startOffset))
+            while (!HasPA30SignatureAtOffset(bytes, startOffset) && !HasPA31SignatureAtOffset(bytes, startOffset))
             {
                 startOffset += 4;
                 if (startOffset >= bytes.Length) throw new InvalidDataException();
             }
             return startOffset;
         }
 
+        private void ParseHeader(BitReader reader,
```
Comment (Wack0) [inline MsDelta/DeltaFile.cs:48]: This function should really be static.

**Comment by m417z** [comment]
For the file attached in #1:
* `HeaderInfoSize` is indeed 0x88.
* `IsPa31` is indeed 1.
* `DeltaClientMinVersion` is 0. Wanna research it a bit more and come up with a more accurate comment?
* `AdditionalHash` is identical to `Hash`.

**Comment by Wack0** [comment]
You're right, DeltaClientMinVersion is 0. I got confused, it's set to 0 but is checked (on read and write) to make sure it's <= 1.

**Comment by m417z** [comment]
I updated the comment.

**Pull request merged**